Sources/PathKit.swift

```diff
@@ -152,8 +152,18 @@ extension Path {
   ///
   public func abbreviate() -> Path {
 #if os(Linux)
-    // TODO: actually de-normalize the path
-    return self
+    let env = ProcessInfo.processInfo.environment
+    guard let home = env["HOME"], !home.isEmpty else { return self }
+    guard let homeRange = self.path.range(of: home, options: [.anchored, .caseInsensitive]) else { return self }
+    let withoutHome = Path(self.path.replacingCharacters(in: homeRange, with: ""))
+    
+    if withoutHome.path.isEmpty || withoutHome.path == "/" {
```

Comment:
Regarding CI for FreeBSD, PS4, other BSDs. I think we can just make our best attempt and writing portable code and leave it down to the users to report problems if PathKit won't compile for them on their platform. They can run the test cases to test compatibility.

I don't think it's possible or feasible for us to be able to CI on every platform.